### ISSUE TITLE
Add optional `annotation` parameter to ivory commands

### ISSUE THREAD

**Issue by EduardSergeev** [ambiata/ivory#720]
Instead of running `annotation update` as a separate add annotation as a part other ivory operation.

Affected commands:
- create-repository
- import-dictionary
- ingest
- snapshot

**Comment by EduardSergeev** [comment]
I was wondering how this "inline annotation" should behave internally. To be specific, I see the following two scenarios:

 1. There is no difference between applying annotation "inline" or doing it as a separate `annotation update` command - the end result, including the internal Ivory data will be identical (i.e. first the main operation is applied, then the annotation is added as a separate commit);
 2. When annotation is applied "inline" with another ivory command both the command's changes and the annotation are recorded in the same commit (i.e. in the final state there will be `-1` less commits in the repository).

I am not sure which way is more in line with the general ivory design, so I need your opinion on this, @markhibberd, @raronson, @charleso.

In terms of implementation `1.` seems to be simple and i like the fact the there will be no difference internally regardless of how we apply annotation (one or two ivory commands). `2.` on the other hand will lead to less frequently incremented commits but will require more changes in codebase and semantically two ways of annotating will be slightly different (internally).

Thoughts? 

PS: The current [WIP](https://github.com/ambiata/ivory/pull/723) takes `2.` approach.

**Comment by markhibberd** [comment]
@EduardSergeev Haven't looked at code yet, but 2 sounds closer to what I would expect - the benefit of the inline one is that it can be attached at the commit it is relevant to, means that the annotations on the operation commit aren't the old ones.

**Comment by EduardSergeev** [comment]
@markhibberd Yes, good point. This way we effectively annotate ivory **operation** instead of adding annotation post-factum to the state of the repository after operation. No chance to get "stale" annotation after that indeed.
